src/vs/base/common/extpath.ts

```diff
@@ -139,9 +139,11 @@ export function isUNC(path: string): boolean {
 }
 
 // Reference: https://en.wikipedia.org/wiki/Filename
-const INVALID_FILE_CHARS = isWindows ? /[\\/:\*\?"<>\|]/g : /[\\/]/g;
 const WINDOWS_FORBIDDEN_NAMES = /^(con|prn|aux|clock\$|nul|lpt[0-9]|com[0-9])$/i;
-export function isValidBasename(name: string | null | undefined): boolean {
+export function isValidBasename(name: string | null | undefined, isWindowsOS?: boolean): boolean {
+	const useWindowsRules = ((isWindowsOS === undefined) ? isWindows : isWindowsOS);
+	const INVALID_FILE_CHARS = useWindowsRules ? /[\\/:\*\?"<>\|]/g : /[\\/]/g;
```

Comment:
@alexr00 suggest to move this back out of the function and make it 2 variables `WINDOWS_INVALID_FILE_CHARS` and `UNIX_INVALID_FILE_CHARS`